pandas/io/sql.py

```diff
@@ -2061,6 +2065,18 @@ def drop_table(self, table_name: str, schema: str | None = None) -> None:
                 self.get_table(table_name, schema).drop(bind=self.con)
             self.meta.clear()
 
+    def delete_rows(self, table_name: str, schema: str | None = None) -> None:
+        schema = schema or self.meta.schema
+        if self.has_table(table_name, schema):
+            self.meta.reflect(
+                bind=self.con, only=[table_name], schema=schema, views=True
+            )
+            with self.run_transaction() as con:
```

Comment:
Hmm I think this implementation is a bit confusing - shouldn't the delete and the append be paired together in a transaction? How is this ensuring the atomicity in its current state?

Reply:
Hello :),

I agree with you but preferred to follow the current coding convention.
The existing implementation  for `replace` was used as the gold standard as it drops & insert new records. Is this assumption flawed ?

I do agree having a single transaction block for both replace and the new `delete_rows` would be more logic / straightforward but again, I tried to follow the convention...

Reply:
Ah OK - fair enough. That seems a little suspect but let's leave that to another cleanup